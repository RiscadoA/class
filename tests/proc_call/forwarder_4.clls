proc overengineered_forward(x:send wait; close, y:recv close; wait){
    cut {
        println("send");
        send ch(x);
        println("lhs fwd");
        fwd ch y
        |ch: recv send wait; close; recv close; wait|
        println("recv");
        recv ch(a);
        println("rhs fwd");
        fwd a ch
    }
};;

proc main() {
    cut {
        println("recv a");
        recv x(a);
        wait x;
        println("close a");
        close a
        |x:send wait; close|
        overengineered_forward(x, y)
        |y:send wait; close|
        send y(a. println("wait a"); wait a; ());
        println("close"); close y
    }
};;
