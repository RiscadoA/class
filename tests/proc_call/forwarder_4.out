close
send
lhs fwd
recv
rhs fwd
recv a
close a
wait a
